src/du_parameters/initial_bwp.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Calculate Initial BWP location And Bandwidth.
+
+Uses subcarrier spacing and carrier bandwidth as inputs.
+"""
+
+import logging
+
+from src.du_parameters.frequency import Frequency
+
+logger = logging.getLogger(__name__)
+
+
+class CalculateBWPLocationBandwidthError(Exception):
+    """Exception raised when calculation of BWP location and bandwidth fails."""
+
+    pass
+
+
+def get_total_prbs_from_scs(subcarrier_spacing: Frequency) -> int:
+    """Get the number of Physical Resource Blocks (PRBs) based on Subcarrier Spacing (SCS).
+
+    Args:
+        subcarrier_spacing (Frequency): Subcarrier Spacing as Frequency object.
+        Valid values: 15, 30, 60, 120.
+
+    Returns:
+        int: Number of PRBs for given Subcarrier spacing value.
+
+    Raises:
+        ValueError: If the Subcarrier spacing value is not supported.
+        TypeError: If the Subcarrier spacing is not a Frequency object.
+    """
+    if not isinstance(subcarrier_spacing, Frequency):
+        raise TypeError(
+            f"Subcarrier spacing must be a Frequency object, not {type(subcarrier_spacing)}"
+        )
+
+    scs_to_prbs = {
+        Frequency.from_khz(15): 275,
+        Frequency.from_khz(30): 137,
+        Frequency.from_khz(60): 69,
+        Frequency.from_khz(120): 33,
+    }
+    try:
+        return scs_to_prbs[subcarrier_spacing]
+    except KeyError:
+        supported_subcarrier_spacing = ", ".join(str(freq) for freq in scs_to_prbs.keys())
```

Comment:
```suggestion
        supported_subcarrier_spacing = ", ".join(str(freq) for freq in scs_to_prbs)
```
Iterating over keys is the default.

Reply:
Done